subsys/bootloader/image/CMakeLists.txt

```diff
@@ -49,14 +49,25 @@ if (DEFINED CONFIG_SOC_NRF9160)
   # Override PM_PROVISION_ADDRESS since Partition Manager cannot yet place
   # partitions in OTP.
   set(otp_start_addr "0xff8108")
+  set(otp_size "760") # See the "UICR" chapter in nRF9160 Product Specification
   set(provision_addr ${otp_start_addr})
+  set(provision_max_size ${otp_size})
 elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
   # Override PM_PROVISION_ADDRESS since Partition Manager cannot yet place
   # partitions in OTP.
   set(otp_start_addr "0xff8100")
+  set(otp_size "768") # See the "UICR" chapter in nRF5340 Product Specification
   set(provision_addr ${otp_start_addr})
+  set(provision_max_size ${otp_size})
 else ()
   set(provision_addr "$<TARGET_PROPERTY:partition_manager,PM_PROVISION_ADDRESS>")
+  set(provision_max_size ${CONFIG_PM_PARTITION_SIZE_PROVISION})
+endif()
+
+if (DEFINED CONFIG_SB_MONOTONIC_COUNTER)
```

Comment:
Could we do the following:

```suggestion
if (DEFINED CONFIG_SB_MONOTONIC_COUNTER)
  set(monotonic_counter_arg "--num-counter-slots ${CONFIG_SB_NUM_MONOTONIC_COUNTERS}")
endif()

# ....

--output ${PROVISION_HEX}
${monotonic_counter_arg}
```

And then ensure the default value is 0 in the .py script?